src/main/scala/is/hail/io/bgen/BgenRecord.scala

```diff
@@ -133,7 +133,7 @@ class BgenRecordV12(compressed: Boolean, nSamples: Int, tolerance: Double) exten
 
     val probabilityIterator = new BgenProbabilityIterator(reader, nBitsPerProb)
 
-    val sampleProbs = new Array[UInt](nGenotypes)
+    val sampleProbs = new Array[Int](nGenotypes)
```

Comment:
Sorry, I should have thought of this when we were chatting, but you can make this type safe again by defining:

```
class ArrayUInt(a: Array[Int]) {
  // only place to use the unchecked methods
  def apply(i: Int): UInt = UInt.applyUnchecked(a(i))
  def update(i: Int, x: UInt) { a(i) = x.intRep }
}
```